File: ejercicios C++/conjetura de collatz.cpp
#include <iostream>

using namespace std;

void conjetura(int x)
{
	cout << x << ",";
	if(x > 1)
	{
		if(x % 2 == 0)
		{
			conjetura(x / 2);
		}
		else
		{
			conjetura(x + x + x + 1);
		}
	}
	
}

int main()
{
	cout << "***PROGRAMA QUE REALIZA LA CONJETURA DE COLLATZ***" << endl;
	
	int x;
	
	cout << " \n Ingrese el numero para empezar la conjetura de collatz: ";
	cin >> x;
	conjetura(x);
	
	
	return 0;
}

File: curso phyton cisco/conjetura de collatz.py
print("Programa que realiza la conjetura de collatz \n")

n = int(input("Ingrese un numero: "))

pasos = 1 

print(n)

while True:
    if n != 1:
        if n % 2 == 0:
            n = float(n / 2)
            print(n)
            pasos += 1
        elif n % 2 != 0:
            n = int((n *3) + 1)
            print(n)
            pasos += 1
    else:
        print()
        print("Numeros de pasos para llegar a 1: ", pasos)
        break